pkg/collector/adapters/config_validate.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapters
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+var (
+	ErrNoPipeline = errors.New("no pipeline available as part of the configuration")
+)
+
+func ConfigValidate(logger logr.Logger, config map[interface{}]interface{}) (map[string]bool, error) {
```

Comment:
Document this function, especially what's the expected output.

Reply:
And contrary to what I mentioned during our conversation, it should probably be a map[interface{}] and not map[string], so that it's consistent with other usages. It would also allow you to skip the cast elsewhere in the code.

Reply:
I've documented the function and set the return as map[interface{}].